src/lib.rs

```diff
@@ -304,6 +304,15 @@ impl Hash for TrashItem {
     }
 }
 
+/// Metadata about a [`TrashItem`]
+#[derive(Debug, Clone)]
+pub struct TrashItemMetadata {
+    /// True if the [`TrashItem`] is a directory, false if it is a file
+    pub is_dir: bool,
+    /// Number of entries for directories, number of bytes for files
+    pub size: u64,
```

Comment:
I think this type makes it too easy to mis-use `size` and treat it as a size of bytes.
Instead, an `enum` could make it absolutely clear what is what, with field names that reflect what's counted.

That enum could then have convenience methods to get the size in bytes for example as `Option<u64>`, and maybe the same for the entries in a directory. Speaking of, I would make clear that this count is non-recursive.

Reply:
I will address this in the next commit